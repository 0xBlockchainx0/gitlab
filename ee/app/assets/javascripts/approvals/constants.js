export const TYPE_USER = 'user';
export const TYPE_GROUP = 'group';
