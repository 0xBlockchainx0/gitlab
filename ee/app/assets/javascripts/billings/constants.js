export const USAGE_ROW_INDEX = 0;
export const BILLING_ROW_INDEX = 1;
export const CUSTOMER_PORTAL_URL = 'https://customers.gitlab.com/subscriptions';
