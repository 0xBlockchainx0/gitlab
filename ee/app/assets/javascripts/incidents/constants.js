import { s__ } from '~/locale';

const tdClass =
  'table-col gl-display-flex d-md-table-cell gl-align-items-center gl-white-space-nowrap';

// eslint-disable-next-line import/prefer-default-export
export const publishedCell = {
  key: 'published',
  label: s__('IncidentManagement|Published'),
  thClass: 'gl-pointer-events-none',
  tdClass,
};
