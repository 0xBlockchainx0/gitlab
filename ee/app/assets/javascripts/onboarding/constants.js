const STORAGE_KEY = 'onboarding_state';

export default STORAGE_KEY;
