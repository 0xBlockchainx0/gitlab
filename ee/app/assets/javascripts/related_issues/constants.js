import { __ } from '~/locale';

export const autoCompleteTextMap = {
  true: {
    issue: __(' or <#issue id>'),
    epic: __(' or <#epic id>'),
  },
  false: {
    issue: '',
    epic: '',
  },
};

export const inputPlaceholderTextMap = {
  issue: __('Paste issue link'),
  epic: __('Paste epic link'),
};

export const relatedIssuesRemoveErrorMap = {
  issue: __('An error occurred while removing issues.'),
  epic: __('An error occurred while removing epics.'),
};

export const pathIndeterminateErrorMap = {
  issue: __('We could not determine the path to remove the issue'),
  epic: __('We could not determine the path to remove the epic'),
};

export const addRelatedIssueErrorMap = {
  issue: __("We can't find an issue that matches what you are looking for."),
  epic: __("We can't find an epic that matches what you are looking for."),
};
