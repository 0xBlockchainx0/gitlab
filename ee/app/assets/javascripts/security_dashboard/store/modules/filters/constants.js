import { s__ } from '~/locale';

export const SEVERITIES = {
  critical: 'Critical',
  high: 'High',
  medium: 'Medium',
  low: 'Low',
  unknown: 'Unknown',
  experimental: 'Experimental',
  ignore: 'Ignore',
  undefined: 'Undefined',
};

export const REPORT_TYPES = {
  sast: s__('ciReport|SAST'),
};
