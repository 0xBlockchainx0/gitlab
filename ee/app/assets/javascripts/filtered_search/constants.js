/* eslint-disable import/prefer-default-export */
import { TOKEN_TYPES as CE_TOKEN_TYPES } from '~/filtered_search/constants';

export const TOKEN_TYPES = [...CE_TOKEN_TYPES, 'approver'];
