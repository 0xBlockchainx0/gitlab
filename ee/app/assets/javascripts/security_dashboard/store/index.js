import Vue from 'vue';
import Vuex from 'vuex';
import configureModerator from './moderator';
import filters from './modules/filters/index';
import projects from './modules/projects/index';
import vulnerabilities from './modules/vulnerabilities/index';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    filters,
    projects,
    vulnerabilities,
  },
});

configureModerator(store);

export default () => store;
