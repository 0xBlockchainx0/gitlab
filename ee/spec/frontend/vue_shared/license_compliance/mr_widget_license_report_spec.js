import Vuex from 'vuex';
import LicenseManagement from 'ee/vue_shared/license_compliance/mr_widget_license_report.vue';
import ReportSection from '~/reports/components/report_section.vue';
import ReportItem from '~/reports/components/report_item.vue';
import { LOADING, ERROR, SUCCESS } from 'ee/vue_shared/security_reports/store/constants';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { TEST_HOST } from 'spec/test_constants';
import {
  approvedLicense,
  blacklistedLicense,
  licenseReport as licenseReportMock,
  generateReportGroup,
} from './mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('License Report MR Widget', () => {
  const apiUrl = `${TEST_HOST}/license_management`;
  const securityApprovalsHelpPagePath = `${TEST_HOST}/path/to/security/approvals/help`;
  let wrapper;

  const defaultState = {
    managedLicenses: [approvedLicense, blacklistedLicense],
    currentLicenseInModal: licenseReportMock[0],
    isLoadingManagedLicenses: true,
  };

  const defaultGetters = {
    isLoading() {
      return false;
    },
    licenseReport() {
      return licenseReportMock;
    },
    licenseSummaryText() {
      return 'FOO';
    },
    reportContainsBlacklistedLicense() {
      return false;
    },
    licenseReportGroups() {
      return [];
    },
  };

  const defaultProps = {
    loadingText: 'LOADING',
    errorText: 'ERROR',
    licensesApiPath: `${TEST_HOST}/parsed_license_report.json`,
    canManageLicenses: true,
    licenseManagementSettingsPath: `${TEST_HOST}/lm_settings`,
    fullReportPath: `${TEST_HOST}/path/to/the/full/report`,
    apiUrl,
    securityApprovalsHelpPagePath,
  };

  const defaultActions = {
    setAPISettings: () => {},
    fetchManagedLicenses: () => {},
    fetchParsedLicenseReport: () => {},
  };

  const mountComponent = ({
    props = defaultProps,
    getters = defaultGetters,
    state = defaultState,
    actions = defaultActions,
  } = {}) => {
    const store = new Vuex.Store({
      modules: {
        licenseManagement: {
          namespaced: true,
          state,
          getters,
          actions,
        },
      },
    });
    return shallowMount(LicenseManagement, {
      localVue,
      propsData: props,
      store,
      stubs: { ReportSection },
    });
  };

  beforeEach(() => {
    wrapper = mountComponent();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('computed', () => {
    describe('hasLicenseReportIssues', () => {
      it('should be false, if the report is empty', () => {
        const getters = {
          ...defaultGetters,
          licenseReport() {
            return [];
          },
        };
        wrapper = mountComponent({ getters });

        expect(wrapper.vm.hasLicenseReportIssues).toBe(false);
      });

      it('should be true, if the report is not empty', () => {
        expect(wrapper.vm.hasLicenseReportIssues).toBe(true);
      });
    });

    describe('licenseReportStatus', () => {
      it('should be `LOADING`, if the report is loading', () => {
        const getters = {
          ...defaultGetters,
          isLoading() {
            return true;
          },
        };
        wrapper = mountComponent({ getters });

        expect(wrapper.vm.licenseReportStatus).toBe(LOADING);
      });

      it('should be `ERROR`, if the report is has an error', () => {
        const state = { ...defaultState, loadLicenseReportError: new Error('test') };
        wrapper = mountComponent({ state });

        expect(wrapper.vm.licenseReportStatus).toBe(ERROR);
      });

      it('should be `SUCCESS`, if the report is successful', () => {
        expect(wrapper.vm.licenseReportStatus).toBe(SUCCESS);
      });
    });

    describe('showActionButtons', () => {
      const { fullReportPath, licenseManagementSettingsPath, ...otherProps } = defaultProps;

      it('should be true if fullReportPath AND licenseManagementSettingsPath prop are provided', () => {
        const props = { ...otherProps, fullReportPath, licenseManagementSettingsPath };
        wrapper = mountComponent({ props });

        expect(wrapper.vm.showActionButtons).toBe(true);
      });

      it('should be true if only licenseManagementSettingsPath is provided', () => {
        const props = { ...otherProps, fullReportPath: null, licenseManagementSettingsPath };
        wrapper = mountComponent({ props });

        expect(wrapper.vm.showActionButtons).toBe(true);
      });

      it('should be true if only fullReportPath is provided', () => {
        const props = {
          ...otherProps,
          fullReportPath,
          licenseManagementSettingsPath: null,
        };
        wrapper = mountComponent({ props });

        expect(wrapper.vm.showActionButtons).toBe(true);
      });

      it('should be false if fullReportPath and licenseManagementSettingsPath prop are not provided', () => {
        const props = {
          ...otherProps,
          fullReportPath: null,
          licenseManagementSettingsPath: null,
        };
        wrapper = mountComponent({ props });

        expect(wrapper.vm.showActionButtons).toBe(false);
      });
    });
  });

  describe('report section', () => {
    it('should render correctly', () => {
      const mockReportGroups = [generateReportGroup()];

      wrapper = mountComponent({
        getters: {
          ...defaultGetters,
          licenseReportGroups() {
            return mockReportGroups;
          },
        },
      });

      expect(wrapper.find(ReportSection).element).toMatchSnapshot();
    });

    it.each`
      givenStatuses                       | expectedNumberOfReportHeadings
      ${[]}                               | ${0}
      ${['failed', 'neutral']}            | ${2}
      ${['failed', 'neutral', 'success']} | ${3}
    `(
      'given reports for: $givenStatuses it has $expectedNumberOfReportHeadings report headings',
      ({ givenStatuses, expectedNumberOfReportHeadings }) => {
        const mockReportGroups = givenStatuses.map(status => generateReportGroup({ status }));

        wrapper = mountComponent({
          getters: {
            ...defaultGetters,
            licenseReportGroups() {
              return mockReportGroups;
            },
          },
        });

        expect(wrapper.findAll({ ref: 'reportHeading' }).length).toBe(
          expectedNumberOfReportHeadings,
        );
      },
    );

    it.each([0, 1, 2])(
      'should include %d report items when section has %# licenses',
      numberOfLicenses => {
        const mockReportGroups = [
          generateReportGroup({
            numberOfLicenses,
          }),
        ];

        wrapper = mountComponent({
          getters: {
            ...defaultGetters,
            licenseReportGroups() {
              return mockReportGroups;
            },
          },
        });

        expect(wrapper.findAll(ReportItem).length).toBe(numberOfLicenses);
      },
    );
  });

  describe('`View full report` button', () => {
    const selector = '.js-full-report';

    it('should be rendered when fullReportPath prop is provided', () => {
      const linkEl = wrapper.find(selector);

      expect(linkEl.exists()).toBe(true);
      expect(linkEl.attributes('href')).toEqual(defaultProps.fullReportPath);
      expect(linkEl.text()).toBe('View full report');
    });

    it('should not be rendered when fullReportPath prop is not provided', () => {
      const props = { ...defaultProps, fullReportPath: null };
      wrapper = mountComponent({ props });

      expect(wrapper.contains(selector)).toBe(false);
    });
  });

  describe('`Manage licenses` button', () => {
    const selector = '.js-manage-licenses';

    it('should be rendered when licenseManagementSettingsPath prop is provided', () => {
      const linkEl = wrapper.find(selector);

      expect(linkEl.exists()).toBe(true);
      expect(linkEl.attributes('href')).toEqual(defaultProps.licenseManagementSettingsPath);
      expect(linkEl.text()).toBe('Manage licenses');
    });

    it('should not be rendered when licenseManagementSettingsPath prop is not provided', () => {
      const props = { ...defaultProps, licenseManagementSettingsPath: null };
      wrapper = mountComponent({ props });

      expect(wrapper.contains(selector)).toBe(false);
    });
  });

  it('should render set approval modal', () => {
    expect(wrapper.find('#modal-set-license-approval')).not.toBeNull();
  });

  it('should init store after mount', () => {
    const actions = {
      setAPISettings: jest.fn(() => {}),
      fetchParsedLicenseReport: jest.fn(() => {}),
    };
    wrapper = mountComponent({ actions });

    expect(actions.setAPISettings).toHaveBeenCalledWith(
      expect.any(Object),
      {
        apiUrlManageLicenses: apiUrl,
        licensesApiPath: defaultProps.licensesApiPath,
        canManageLicenses: true,
      },
      undefined,
    );

    expect(actions.fetchParsedLicenseReport).toHaveBeenCalledWith(
      expect.any(Object),
      undefined,
      undefined,
    );
  });

  describe('approval status', () => {
    const findSecurityApprovalHelpLink = () => wrapper.find('.js-security-approval-help-link');

    it('does not show a link to security approval help page if report does not contain blacklisted licenses', () => {
      expect(findSecurityApprovalHelpLink().exists()).toBe(false);
    });

    it('shows a link to security approval help page if report contains blacklisted licenses', () => {
      const getters = {
        ...defaultGetters,
        reportContainsBlacklistedLicense() {
          return true;
        },
      };
      wrapper = mountComponent({ getters });
      const securityApprovalHelpLink = findSecurityApprovalHelpLink();

      expect(findSecurityApprovalHelpLink().exists()).toBe(true);
      expect(securityApprovalHelpLink.attributes('href')).toBe(securityApprovalsHelpPagePath);
    });
  });
});
