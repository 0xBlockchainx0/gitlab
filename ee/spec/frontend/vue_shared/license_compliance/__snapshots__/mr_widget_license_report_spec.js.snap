// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`License Report MR Widget report section should render correctly 1`] = `
<section
  class="media-section license-report-widget mr-report"
  data-qa-selector="license_report_widget"
>
  <div
    class="media"
  >
    <status-icon-stub
      class="align-self-center"
      size="24"
      status="warning"
    />
     
    <div
      class="media-body d-flex flex-align-self-center align-items-center"
    >
      <div
        class="js-code-text code-text"
      >
        <div>
          
          
          
      FOO
      
          <!---->
           
          <!---->
        </div>
         
      </div>
       
      <div
        class="append-right-default"
      >
        <a
          class="btn btn-default btn-sm js-manage-licenses append-right-8"
          href="http://test.host/lm_settings"
        >
          
        Manage licenses
      
        </a>
         
        <a
          class="btn btn-default btn-sm js-full-report"
          href="http://test.host/path/to/the/full/report"
          target="_blank"
        >
          
        View full report 
          <icon-stub
            name="external-link"
            size="16"
          />
        </a>
      </div>
       
      <button
        class="js-collapse-btn btn float-right btn-sm align-self-center qa-expand-report-button"
        type="button"
      >
        
        Expand
      
      </button>
    </div>
  </div>
   
  <div
    class="js-report-section-container"
    style="display: none;"
  >
    <smart-virtual-list-stub
      class="report-block-container"
      length="1"
      remain="20"
      rtag="div"
      size="26"
      wclass="report-block-list mb-2"
      wtag="ul"
    >
      <li
        class="ml-2 mb-1"
      >
        <h2
          class="h5 m-0"
        >
          some-status
        </h2>
         
        <p
          class="m-0"
        >
          some-status
        </p>
      </li>
       
    </smart-virtual-list-stub>
  </div>
</section>
`;
